Sources/EdgeNetworkHandlers/NetworkResponseHandler.swift

```diff
@@ -33,13 +36,12 @@ class NetworkResponseHandler {
     func addWaitingEvents(requestId: String, batchedEvents: [Event]) {
         guard !requestId.isEmpty, !batchedEvents.isEmpty else { return }
 
-        let eventIds = batchedEvents.map { $0.id.uuidString }
         serialQueue.sync {
             if self.sentEventsWaitingResponse[requestId] != nil {
                 Log.warning(label: self.LOG_TAG, "Name collision for requestId \(requestId), events list is overwritten.")
             }
 
-            self.sentEventsWaitingResponse[requestId] = eventIds
+            self.sentEventsWaitingResponse[requestId] = batchedEvents
```

Comment:
we might need to optimize this and only keep the event id and timestamp. if there are many events with large payload being processed at a time, this can bump the memory usage quickly.

Reply:
changed this to store tuples of `(uuid: String, date: Date)`